src/rules/noUnspecifiedLiteral.ts

```diff
@@ -0,0 +1,48 @@
+import * as es from 'estree'
+
+import { ErrorSeverity, ErrorType, Rule, SourceError } from '../types'
+
+const specifiedLiterals = ['boolean', 'string', 'number']
+
+export class NoUnspecifiedLiteral implements SourceError {
+  public type = ErrorType.SYNTAX
+  public severity = ErrorSeverity.ERROR
+
+  constructor(public node: es.Literal) {}
+
+  get location() {
+    return this.node.loc!
+  }
+
+  public explain() {
+    /** 
+     * A check is used for RegExp to ensure that only null and RegExp are caught.
+     * Any other unspecified literal value should not be caught.
+     */
+    const literal = this.node.value === null ? 'null' 
+      : this.node.value instanceof RegExp ? 'RegExp'
+      : ''
+    return (
+      `A literal '${literal}' value is not allowed. Only string, boolean, and number literal values are allowed.`
```

Comment:
Prefer `'${literal} literals are not allowed'`, to be consistent with the other disallowed stuff, .e.g

> Line 1: Array expressions are not allowed

Reply:
@ningyuansg Agreed. Updated!